tests/Library/Vanilla/EmbeddedContent/Factories/PanoptoEmbedFactoryTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * @author Patrick Desjardins <patrick.d@vanillaforums.com>
+ * @copyright 2009-2020 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace VanillaTests\Library\Vanilla\EmbeddedContent\Factories;
+
+use Vanilla\EmbeddedContent\Embeds\PanoptoEmbed;
+use Vanilla\EmbeddedContent\Factories\PanoptoEmbedFactory;
+use VanillaTests\MinimalContainerTestCase;
+
+/**
+ * Tests for the Panopto embed factory.
+ */
+class PanoptoEmbedFactoryTest extends MinimalContainerTestCase {
+
+    /** @var PanoptoEmbedFactory */
+    private $factory;
+
+    /**
+     * Set the factory.
+     */
+    public function setUp(): void {
+        parent::setUp();
+        $this->factory = new PanoptoEmbedFactory();
+    }
+
+    /**
+     * Test that all domain types are supported.
+     *
+     * @param string $urlToTest
+     * @dataProvider supportedUrlsProvider
+     */
+    public function testSupportedUrls(string $urlToTest) {
+        $this->assertTrue($this->factory->canHandleUrl($urlToTest));
+    }
+
+    /**
+     * Return an array of supported urls.
+     *
+     * @return array
+     */
+    public function supportedUrlsProvider(): array {
+        return [
+            ['https://howtovideos.hosted.panopto.com/Panopto/Pages/Viewer.aspx?id=a5f152d0-0718-45de-bab0-a937015c2f35'],
+            ['https://demo.ca.panopto.com/Panopto/Pages/Viewer.aspx?id=7604d3f5-c51d-4053-8c59-ab4400e0bbe6'],
+            ['https://demo.cloud.panopto.eu/Panopto/Pages/Viewer.aspx?id=2eb0bf71-e051-4386-8aa3-0c7cf8a28135'],
+            ['https://demo.ap.panopto.com/Panopto/Pages/Viewer.aspx?id=25329997-baa2-42cf-89c4-ab4400e080e3'],
+        ];
+    }
+
+    /**
+     * Test the Panopto Embed instantiation.
+     */
+    public function testCreateEmbedForUrl() {
```

Comment:
Structurally it's preferred to use a PHPUnit provider for this type of thing.

https://github.com/vanilla/vanilla/blob/0359e395815443b113a7f88a4b9ba8f02a5d8e25/tests/Library/Vanilla/EmbeddedContent/Factories/ScrapeEmbedFactoryTest.php#L41-L53

Reply:
On second thought, you're already using a data provider in your earlier test. Couldn't you wire this one up the same way?

Reply:
Yes I can do that.